sopel/logger.py

```diff
@@ -30,40 +34,108 @@ def formatException(self, exc_info):
+                'level': base_level,
+                'handlers': ['console', 'logfile', 'errorfile'],
+            },
+            # raw IRC log
+            'sopel.raw': {
+                'level': 'DEBUG',
+                'propagate': False,
+                'handlers': ['raw'],
+            },
+            # asynchat exception logger
+            'sopel.exceptions': {
+                'level': 'INFO',
+                'propagate': False,
+                'handlers': ['exceptionfile'],
+            },
+        },
+        'handlers': {
+            # output on stderr
+            'console': {
+                'level': 'DEBUG',
+                'class': 'logging.StreamHandler',
+                'formatter': 'sopel',
+            },
+            # generic purpose log file
+            'logfile': {
+                'level': 'DEBUG',
+                'class': 'logging.handlers.TimedRotatingFileHandler',
+                'filename': os.path.join(
+                    log_directory, settings.basename + '.sopel.log'),
+                'when': 'midnight',
+                'formatter': 'sopel',
+            },
+            # catched error log file
+            'errorfile': {
+                'level': 'ERROR',
+                'class': 'logging.handlers.TimedRotatingFileHandler',
+                'filename': os.path.join(
+                    log_directory, settings.basename + '.error.log'),
+                'when': 'midnight',
+                'formatter': 'sopel',
+            },
+            # uncaught error file
+            'exceptionfile': {
+                'level': 'ERROR',
+                'class': 'logging.handlers.TimedRotatingFileHandler',
+                'filename': os.path.join(
+                    log_directory, settings.basename + '.exceptions.log'),
+                'when': 'midnight',
+                'formatter': 'sopel',
+            },
+            # raw IRC log file
+            'raw': {
+                'level': 'DEBUG',
+                'class': 'logging.handlers.TimedRotatingFileHandler',
+                'filename': os.path.join(
+                    log_directory, settings.basename + '.raw.log'),
+                'when': 'midnight',
+                'formatter': 'raw',
+            },
+        },
+    }
+    dictConfig(logging_config)
 
 
 def get_logger(name=None):
     """Return a logger for a module, if the name is given.
 
-    This is equivalent to `logging.getLogger('sopel.modules.' + name)` when
-    name is given, and `logging.getLogger('sopel')` when it is not. The latter
-    case is intended for use in Sopel's core; modules should call
-    `get_logger(__name__)` to get a logger."""
-    if name:
-        return logging.getLogger('sopel.modules.' + name)
-    else:
+    .. deprecated:: 7.0
+
+        Use ``logging.getLogger(__name__)`` in Sopel's code instead, and
+        :func:`sopel.tools.get_logger` for external plugins.
+
+        This will warn a deprecation warning in Sopel 8.0 then removed in 9.0.
```

Comment:
Why not add `@deprecated` (the warning) now, and remove it in 8.0? Use this suggested change if appropriate:

```suggestion
        Will be removed in Sopel 8.0.
```

Reply:
Because I assume a lot of external plugins are using it right now, so it'd raise a *lot* of warnings. I'm not sure it's a good idea, but I don't have a strong opinion.

If that was me, deprecation warning would be all over the place... :stuck_out_tongue:

Reply:
Still gonna mention it specifically in the migration guide for 7, then.